python/gui/MaaS/cbv/DatasetManagementView.py

```diff
@@ -0,0 +1,114 @@
+"""
+Defines a view that may be used to configure a MaaS request
+"""
+import asyncio
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+
+import dmod.communication as communication
+from dmod.core.meta_data import DataCategory, DataFormat
+
+import logging
+logger = logging.getLogger("gui_log")
+
+from .utils import extract_log_data
+from .AbstractDatasetView import AbstractDatasetView
+
+
+class DatasetManagementView(AbstractDatasetView):
+
+    """
+    A view used to configure a dataset management request or requests for transmitting dataset data.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _process_event_type(self, http_request: HttpRequest) -> communication.MessageEventType:
+        """
+        Determine and return whether this request is for a ``DATASET_MANAGEMENT`` or ``DATA_TRANSMISSION`` event.
+
+        Parameters
+        ----------
+        http_request : HttpRequest
+            The raw HTTP request in question.
+
+        Returns
+        -------
+        communication.MessageEventType
+            Either ``communication.MessageEventType.DATASET_MANAGEMENT`` or
+            ``communication.MessageEventType.DATA_TRANSMISSION``.
+        """
+        # TODO:
+        raise NotImplementedError("{}._process_event_type not implemented".format(self.__class__.__name__))
+
+    def get(self, http_request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        """
+        The handler for 'get' requests.
+
+        This will render the 'maas/dataset_management.html' template after retrieving necessary information to initially
+        populate the forms it displays.
+
+        Parameters
+        ----------
+        http_request : HttpRequest
+            The request asking to render this page.
+        args
+        kwargs
+
+        Returns
+        -------
+        A rendered page.
+        """
+        errors, warnings, info = extract_log_data(kwargs)
+
+        # Gather map of serialized datasets, keyed by dataset name
+        serial_dataset_map = asyncio.get_event_loop().run_until_complete(self.get_datasets())
+        serial_dataset_list = [serial_dataset_map[d] for d in serial_dataset_map]
```

Comment:
this is nit-picky, but for code readability, `serial_dataset_list = list( serial_dataset_map.values())` might be preferred.